cpp/src/arrow/dataset/dataset_writer.cc

```diff
@@ -476,7 +478,8 @@ class DatasetWriter::DatasetWriterImpl : public util::AsyncDestroyable {
   }
 
   Future<> DoWriteRecordBatch(std::shared_ptr<RecordBatch> batch,
-                              const std::string& directory, const std::string& prefix) {
+                              const std::string& directory, const std::string& prefix,
+                              const bool& create_dir) {
```

Comment:
No need to pass primitive types by reference.
```suggestion
                              bool create_dir) {
```

Reply:
Also, I may be misunderstanding, but this parameter doesn't seem actually used?

Reply:
yes, sorry we don't need to pass it by reference. Yes, the parameter is actually needed. Thanks for pointing that out, thanks!